onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -67,46 +67,55 @@
     ComPtr<ID3D12RootSignature> m_rootSignature;
     ComPtr<ID3D12PipelineState> m_pipelineState;
 
-    std::vector<uint32_t> m_inputDims = {};
-    std::vector<uint32_t> m_outputDims = {};
     int64_t m_axis;
     bool m_isOnesided;
     bool m_isInverse;
 
-    uint32_t m_outputDataSize = 0;
-    uint32_t m_inputDataSize = 0;
-    uint32_t m_outputIdx = 0;
-    uint32_t m_numPasses = 0;
-
-    // Allocate temporary buffers if needed
-    struct ResourceDesc
-    {
-        ComPtr<ID3D12Resource> Resource;
-        std::array<uint32_t, 4> Sizes;
-        std::array<uint32_t, 4> Strides;
-    };
-    std::vector<ResourceDesc> m_resourceLoopList = {};
+    struct StockhamParameters {
+        // Allocate temporary buffers if needed
+        struct ResourceDesc
+        {
+            ComPtr<ID3D12Resource> Resource;
+            std::array<uint32_t, 4> Sizes;
+            std::array<uint32_t, 4> Strides;
+        };
 
-    struct LoopRange
-    {
-        unsigned Left;
-        unsigned Right;
-        unsigned End;
-        unsigned CalculateIndex(unsigned index)
+        struct LoopRangeCalculator
         {
-            if (index > 0 && index < End)
-            {
-                unsigned range = Right - Left + 1;
-                index = Left + (index - 1) % range;
-            }
-            else if (index == End)
+            unsigned Left;
+            unsigned Right;
+            unsigned End;
+            unsigned CalculateIndex(unsigned index)
             {
-                index = Right + 1;
+                if (index > 0 && index < End)
+                {
+                    unsigned range = Right - Left + 1;
+                    index = Left + (index - 1) % range;
+                }
+                else if (index == End)
+                {
+                    index = Right + 1;
+                }
+                return index;
             }
-            return index;
-        }
+        };
+
+        std::vector<ResourceDesc> ResourceLoopList = {};
+        LoopRangeCalculator LoopRange = {};
+        uint32_t OutputIndex = 0;
+        uint32_t NumberOfPasses = 0;
+    };
+
+    enum class DFTType {
+        Stockham = 0,
+        BluesteinZChirp,
+    };
+
+    struct DFTParameters {
```

Comment:
> { [](http://example.com/codeflow?start=24&length=2)

[nit] internal consistency of brace style with existing places like line 121. Applies to line 109 and 74 too.

Reply:
done